server/pbench/bin/index-pbench

```diff
@@ -241,12 +242,43 @@ class PbenchData(object):
         # string timestamp format.
         ts_float = float(orig_ts)/1000
         ts = datetime.utcfromtimestamp(ts_float)
-        if self.start_run_ts is not None and ts < self.start_run_ts:
-            self.counters['result_ts_before_start_run_ts'] += 1
-        if self.end_run_ts is not None and ts > self.end_run_ts:
-            self.counters['result_ts_after_end_run_ts'] += 1
+        if self.ptb.start_run_ts is not None and ts < self.ptb.start_run_ts:
+            # The calculated timestamp, `ts`, is earlier that the timestamp of
+            # the start of the pbench run itself, `start_run_ts`.  That can
+            # occur because of at least 3 different situations: 1. the
+            # recorded start run timestamp, `start_run_ts`, is not correct;
+            # 2. the `ts` is a valid absolute timestamp that was incorrectly
+            # recorded; 3. the `ts` is a valid _relative_ timestamp, measured
+            # from the beginning of some unknown time `T0`.
+            #
+            # Since we do not have enough information at this point to correct
+            # or detect the 1st or 2nd situation, we make an attempt to detect
+            # and correct the 3rd situation.
+            #
+            # For the 3rd situation, we make an attempt to treat the original
+            # timestamp as if it is relative to the start of the run, and
+            # calculate a _new_ absolute timestamp from that point.  If that
+            # new absolute timestamp is before the end of the run as well, we
+            # just use the new absolute timestamp instead of the relative
+            # timestamp.  If the new absolute timestamp is beyond the end of
+            # the run, it is likely that the timestamp is not relative so we
+            # use it as is.
```

Comment:
The flaw in this logic is that it is determined in isolation each time.  We might want to determine this once and decide to apply it consistently for all related timestamps.